### PR TITLE
fix(accessibility): tabbing and focusing on chat page #ACC-16

### PR DESCRIPTION
**Pull request by kamil-au**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-16" title="ACC-16" target="_blank">SubtaskACC-16</a>  9.2.1.1 Fix keyboard focus on text chat page
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Fixing keyboard focus on the main chat page

### CODE REVIEW

File: src/page/template/detail-view.htm [hunk 1]
```diff
@@ -6,29 +6,31 @@
...
         </div>
-        <close-icon class="detail-view-header-close-button icon-button" data-bind="click: clickOnClose" data-uie-name="do-close-detail-view"></close-icon>
+        <button type="button" class="detail-view-header-close-button icon-button" data-bind="click: clickOnClose, attr: {'aria-label': t('conversationDetailsCloseLabel')}" data-uie-name="do-close-detail-view">
+          <close-icon></close-icon>
+        </button>
       </header>
-      <div class="detail-view-main" data-bind="click: clickOnClose">
-        <img class="detail-view-image" data-bind="attr: {src: imageSrc()}" data-uie-name="status-picture"/>
+      <div class="detail-view-main" data-bind="click: clickOnClose, event: { keydown: (_, event) => hadndlePressOnClose(event) }, attr: {'aria-label': t('conversationDetailsCloseLabel')}" tabindex="0" role="button">
+        <img class="detail-view-image" data-bind="attr: {src: imageSrc()}" data-uie-name="status-picture" alt=""/>
```
Comment (tlebon) [inline src/page/template/detail-view.htm:14]: maybe the alt should share the data-uie-name value?

File: src/script/page/collection/fullSearch/FullSearchItem.tsx [hunk 2]
```diff
@@ -36,8 +37,23 @@ const FullSearchItem: React.FC<FullSearchItemProps> = ({message, onClick, format
   const {user, timestamp} = useKoSubscribableChildren(message, ['user', 'timestamp']);
   const {name} = useKoSubscribableChildren(user, ['name']);
   const {parts, matches} = formatText((message.getFirstAsset() as Text).text);
+
+  const handleSearchItemKeydown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (tlebon) [inline src/script/page/collection/fullSearch/FullSearchItem.tsx]: this same (more or less) function appears a few times, can we move it to the keyboard utils?

File: src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset.tsx [hunk 3]
```diff
@@ -80,6 +82,17 @@ const ImageAsset: React.FC<ImageAssetProps> = ({asset, message, onClick, teamSta
 
   const dummyImageUrl = `data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1 1' width='${asset.width}' height='${asset.height}'></svg>`;
 
+  const handleImageKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```
Comment (tlebon) [inline src/script/components/MessagesList/Message/ContentMessage/asset/ImageAsset.tsx]: other instance

File: src/script/components/MessagesList/Message/ContentMessage/asset/FileAssetComponent.tsx [hunk 4]
```diff
@@ -67,6 +68,13 @@ const FileAssetComponent: React.FC<FileAssetProps> = ({
   const isDownloading = assetStatus === AssetTransferState.DOWNLOADING;
   const isUploading = assetStatus === AssetTransferState.UPLOADING;
 
+  const handleKeyPress = (event: React.KeyboardEvent) => {
```
Comment (tlebon) [inline src/script/components/MessagesList/Message/ContentMessage/asset/FileAssetComponent.tsx]: other instance

File: src/script/page/collection/fullSearch/FullSearchItem.tsx (hunk 2 above)
Comment (tlebon) [reply src/script/page/collection/fullSearch/FullSearchItem.tsx]: maybe just receives the function to be called as the second param

File: src/page/template/detail-view.htm (hunk 1 above)
Comment (kamil-au) [reply src/page/template/detail-view.htm:14]: Empty alt is ok here, as this image is only a presentation one. Additionally, we have a "button" with a label on it above so the reader will read this label :)

File: src/script/view_model/ImageDetailViewViewModel.ts [hunk 5]
```diff
@@ -178,6 +180,12 @@ export class ImageDetailViewViewModel {
     });
   };
 
+  readonly hadndlePressOnClose = (event: KeyboardEvent): void => {
```
Comment (tlebon) [inline src/script/view_model/ImageDetailViewViewModel.ts]: these two methods are also the same.. maybe they can be abstracted also? 🤷‍♂️

Comment (tlebon) [reply src/script/view_model/ImageDetailViewViewModel.ts]: typo on the naming here though! 
```suggestion
  readonly handlePressOnClose = (event: KeyboardEvent): void => {
```

File: src/script/view_model/content/TitleBarViewModel.ts [hunk 6]
```diff
@@ -197,6 +198,12 @@ export class TitleBarViewModel {
     this.showDetails(false);
   };
 
+  readonly pressOnDetails = (viewMode: unknown, event: KeyboardEvent): void => {
```
Comment (tlebon) [inline src/script/view_model/content/TitleBarViewModel.ts]: other similar method.. why is this one receiving `viewMode` also? seems unused

File: src/script/util/KeyboardUtil.ts [hunk 7]
```diff
@@ -136,6 +136,13 @@ export const offEscKey = (handler: KeyboardHandler) => {
   }
 };
 
+export const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>, callback: () => void) => {
+  if (event.key === KEY.ENTER || event.key === KEY.SPACE) {
```
Comment (tlebon) [inline src/script/util/KeyboardUtil.ts:140]: maybe should also use the `isSpaceKey(event) || isEnterKey(event)` syntax of the methods? but im okay with this also

**Review by tlebon — APPROVED** [review]

**Review by atomrc — APPROVED** [review]

**Pull request merged**